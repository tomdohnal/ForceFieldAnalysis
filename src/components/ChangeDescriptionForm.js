// @flow
import React from 'react';
import { Form, TextArea, Button } from 'semantic-ui-react';

type Props = {
  onSubmit: () => void,
  onInputChange: (e: SyntheticEvent<HTMLInputElement>, value: { value: string }) => void,
  inputValue: ?string,
}

const ChangeDescriptionForm = ({
  onSubmit, onInputChange, inputValue,
}: Props) => (
  <Form onSubmit={onSubmit}>
    <Form.Group>
      <TextArea
        autoHeight
        onChange={onInputChange}
        placeholder="Enter the description of your change (optional)"
        value={inputValue}
      />
    </Form.Group>
    <Form.Group>
      <Button>Submit Description</Button>
      <Button compact size="mini">No description</Button>
    </Form.Group>
  </Form>
);

export default ChangeDescriptionForm;
