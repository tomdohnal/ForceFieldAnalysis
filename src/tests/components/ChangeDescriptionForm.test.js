import React from 'react';
import { shallow } from 'enzyme';

import ChangeDescriptionForm from '../../components/ChangeDescriptionForm';

describe('ChangeDescriptionForm', () => {
  const onSubmit = jest.fn();
  const onInputChange = jest.fn();

  const ShallowChangeDescriptionForm = shallow(
    <ChangeDescriptionForm
      inputValue="the description of the force"
      onSubmit={onSubmit}
      onInputChange={onInputChange}
    />,
  );

  describe('rendering', () => {
    it('renders correctly', () => {
      expect(ShallowChangeDescriptionForm).toMatchSnapshot();
    });
  });

  describe('functions triggering', () => {
    it('fires onSubmit function when the form is submitted', () => {
      ShallowChangeDescriptionForm.dive().find('form').simulate('submit');

      expect(onSubmit).toBeCalled();
    });

    it('fires onInputChange event when the input is changed', () => {
      ShallowChangeDescriptionForm.find('TextArea').dive().find('textarea').simulate('change');

      expect(onInputChange).toBeCalled();
    });
  });
});
