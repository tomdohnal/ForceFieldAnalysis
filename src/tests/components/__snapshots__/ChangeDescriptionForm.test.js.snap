// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeDescriptionForm rendering renders correctly 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormGroup>
    <TextArea
      as="textarea"
      autoHeight={true}
      onChange={[Function]}
      placeholder="Enter the description of your change (optional)"
      rows={3}
      value="the description of the force"
    />
  </FormGroup>
  <FormGroup>
    <Button
      as="button"
    >
      Submit Description
    </Button>
    <Button
      as="button"
      compact={true}
      size="mini"
    >
      No description
    </Button>
  </FormGroup>
</Form>
`;
